### ISSUE TITLE
Investigate physics-informed neural network (PINN) training strategies

### ISSUE THREAD

**Issue by ChrisRackauckas** [SciML/NeuralPDE.jl#71]
https://github.com/JuliaDiffEq/NeuralNetDiffEq.jl/issues/46 shows that the basic strategy of training on a fixed training grid doesn't work so well even on a 1-dimensional PINN (an ODE) if the nonlinearity is strong enough. To improve the training process, we should look at changing the strategies a bit. Here's some things that come to mind:

- [x] Random sampling in the domain, making a stochastic loss function to prevent local minima
- [x] Random minibatching
- [x] Starting the training with a small time interval, and successively growing it (user side)
- [x] Growing quasi-Monte Carlo sampling (https://github.com/JuliaDiffEq/QuasiMonteCarlo.jl) (done https://github.com/SciML/NeuralPDE.jl/pull/189)

A new type should be added to the `alg` to allow dispatching on `strategy`.
